### PR TITLE
Cleanup and improve existing docs, part 2

### PR DESCRIPTION
**Pull request by designatednerd**
In this PR: 
- Moved all information related to Caching into a `caching` document for clarity
- Added some advanced codegen tips and tricks to `installation`
- Added links to where stuff was moved
- Updated order of sections in sidebar for a better order to read through conceptually
- Fixed a hot mess of broken links in the auto-generated documentation

### CODE REVIEW

**Comment by designatednerd** [comment]
Gonna leave this up until lunch (🇺🇸-central time) tomorrow for comments and feedback from the couple of issues I pinged about these updates.

**Comment by designatednerd** [comment]
OK, lunch eaten! If you've still got questions or comments on this, please feel free to leave 'em and I'll try to address in a future PR.

File: docs/source/caching.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+All caches can be cleared in their entirety by calling [`clear(callbackQueue:completion:)`](api/Apollo/protocols/NormalizedCache/#clearcallbackqueuecompletion). If you need to work more directly with the cache, please see the [Direct Cache Access](#direct-cache-access) section.
+
+## Controlling normalization
+
+While Apollo can do basic caching based on the shape of GraphQL queries and their results, Apollo won't be able to associate objects fetched by different queries without additional information about the identities of the objects returned from the server. 
+
+This is referred to as [cache normalization](http://dev.apollodata.com/core/how-it-works.html#normalize). You can read about our caching model in detail in our blog post, ["GraphQL Concepts Visualized"](https://medium.com/apollo-stack/the-concepts-of-graphql-bc68bd819be3).
```
Comment (martijnwalraven) [inline docs/source/caching.md]: This still links to dev.apollodata.com.

Comment (designatednerd) [reply docs/source/caching.md]: womp. Good catch. I'll see where that moved to

Comment (designatednerd) [reply docs/source/caching.md]: Updated directly on master :trollface:

**Pull request merged**